src/SportsHub.Infrastructure/Repositories/TeamRepository.cs

```diff
@@ -79,5 +79,35 @@ public async Task<bool> DoesTeamAlredyExistByIdAsync(Guid id)
 
             return teams.Any(team => team.Id == id);
         }
+
+        public IEnumerable<Team> GetTeamsFilteredByLocation(string location, ICollection<Team> teams)
+        {
+            return teams.Where(teams => teams.Location == location).ToList();
+        }
+
+        public IEnumerable<Team> GetTeamsFilteredBySubcategoryId(Guid subcategoryId, ICollection<Team> teams)
+        {
+            return teams.Where(teams => teams.SubcategoryId == subcategoryId).ToList();
+        }
+
+        public IEnumerable<Team> GetTeamsFilteredBySubcategoryIds(IQueryable<Subcategory> subcategories, ICollection<Team> teams)
```

Comment:
Sorry, but I didn't get the intent of the method.

Reply:
I've created two methods with similar names: GetTeamsFilteredBySubcategoryId and GetTeamsFilteredBySubcategoryIds, because I wanted to filter teams by subcategory name and by category name respectively.
In the first case: you have one subcategory id to filter teams.
In the second case: you have one category id, to find all subcategories in this category to filter all teams in those subcategories.
In this second case were some troubles with multichoosing teams by different subcategories, so I've created another method which I called "GetTeamsFilteredBySubcategoryIds" to implement this.